StabilityMatrix.Core/Models/Packages/FluxGym.cs

```diff
@@ -0,0 +1,178 @@
+using StabilityMatrix.Core.Services;
+
+namespace StabilityMatrix.Core.Models.Packages;
+
+[Singleton(typeof(BasePackage))]
+public class FluxGym(
+    IGithubApiCache githubApi,
+    ISettingsManager settingsManager,
+    IDownloadService downloadService,
+    IPrerequisiteHelper prerequisiteHelper
+)
+    : BaseGitPackage(githubApi, settingsManager, downloadService, prerequisiteHelper),
+        ISharedFolderLayoutPackage
+{
+    public override string Name => "FluxGym";
+    public override string DisplayName { get; set; } = "FluxGym";
+    public override string Author => "cocktailpeanut";
+
+    public override string Blurb => "Dead simple FLUX LoRA training UI with LOW VRAM support";
+
+    public override string LicenseType => "N/A";
+    public override string LicenseUrl => "";
+    public override string LaunchCommand => "app.py";
+
+    public override Uri PreviewImageUri =>
+        new("https://raw.githubusercontent.com/cocktailpeanut/fluxgym/main/screenshot.png");
+
+    public override List<LaunchOptionDefinition> LaunchOptions => [LaunchOptionDefinition.Extras];
+
+    public override SharedFolderMethod RecommendedSharedFolderMethod => SharedFolderMethod.Configuration;
+
+    public override IEnumerable<SharedFolderMethod> AvailableSharedFolderMethods =>
+        new[] { SharedFolderMethod.Symlink, SharedFolderMethod.Configuration, SharedFolderMethod.None };
+
+    public override Dictionary<SharedFolderType, IReadOnlyList<string>> SharedFolders =>
+        ((ISharedFolderLayoutPackage)this).LegacySharedFolders;
+
+    public virtual SharedFolderLayout SharedFolderLayout =>
+        new()
+        {
+            RelativeConfigPath = "config.txt",
+            Rules =
+            [
+                new SharedFolderLayoutRule
+                {
+                    SourceTypes = [SharedFolderType.CLIP],
+                    TargetRelativePaths = ["models/clip"]
+                },
+                new SharedFolderLayoutRule
+                {
+                    SourceTypes = [SharedFolderType.Unet],
+                    TargetRelativePaths = ["models/unet"]
+                },
+                new SharedFolderLayoutRule
+                {
+                    SourceTypes = [SharedFolderType.VAE],
+                    TargetRelativePaths = ["models/vae"],
+                    ConfigDocumentPaths = ["path_vae"]
+                },
+                new SharedFolderLayoutRule
+                {
+                    TargetRelativePaths = [OutputFolderName],
+                    ConfigDocumentPaths = ["path_outputs"]
+                }
+            ]
+        };
+
+    public override Dictionary<SharedOutputType, IReadOnlyList<string>> SharedOutputFolders =>
+        new() { [SharedOutputType.Text2Img] = new[] { "outputs" } };
+
+    public override IEnumerable<TorchIndex> AvailableTorchIndices =>
+        new[] { TorchIndex.Cpu, TorchIndex.Cuda, TorchIndex.DirectMl, TorchIndex.Rocm, TorchIndex.Mps };
+
```

Comment:
Are these actually supported? The only instructions on the repo I can find is for nvidia, although the pinokio scripts seem to have some stuff for ROCm / MPS. 

If it's just intended to be for nvidia users, this'll need to change, otherwise if these are actually all supported, you'll need to handle installing the correct version of torch based on the `options.PythonOptions.TorchIndex` in the `InstallPackage` method.

Reply:
I didn't test the pinokio scripts so yes, this should only be cuda! Ill change that